File: suhyeok/완전탐색/moigosa.py
def solution(answers):
    answer = []

    # 1번 학생
    student_answer1 = [1, 2, 3, 4, 5]
    # 2번 학생
    student_answer2 = [2, 1, 2, 3, 2, 4, 2, 5]
    # 3번 학생
    student_answer3 = [3, 3, 1, 1, 2, 2, 4, 4, 5, 5]

    # 학생들 맞은 갯수 세는 배열
    cnt = [0, 0, 0]

    for i in range(len(answers)):
        if answers[i] == student_answer1[i % len(student_answer1)]:
            cnt[0] += 1
        if answers[i] == student_answer2[i % len(student_answer2)]:
            cnt[1] += 1
        if answers[i] == student_answer3[i % len(student_answer3)]:
            cnt[2] += 1

    max_score = max(cnt)

    for i in range(0, 3):
        if max_score == cnt[i]:
            answer.append(i+1)

    return answer

File: pkeugine/Moigosa.java
import java.util.ArrayList;
import java.util.List;

// 프로그래머스
// 모의고사
public class Moigosa {

    public static void main(String[] args) {
        Solution sol = new Solution();

        int[] firstInput = {1,2,3,4,5};
        int[] firstAnswer = sol.solution(firstInput);

        int[] secondInput = {1,3,2,4,2};
        int[] secondAnswer = sol.solution(secondInput);

        System.out.println("first answer");
        for (int i = 0; i < firstAnswer.length; i++) {
            System.out.println(firstAnswer[i]);
        }

        System.out.println("second answer");
        for (int i = 0; i < secondAnswer.length; i++) {
            System.out.println(secondAnswer[i]);
        }
    }
}

class Solution {

    public int[] solution(int[] answers) {
        int[] a = {1, 2, 3, 4, 5};
        int[] b = {2, 1, 2, 3, 2, 4, 2, 5};
        int[] c = {3, 3, 1, 1, 2, 2, 4, 4, 5, 5};
        int[] score = new int[3];

        int answersLength = answers.length;
        int aLength = a.length;
        int bLength = b.length;
        int cLength = c.length;
        for (int i = 0; i < answersLength; i++) {
            if (answers[i] == a[i%aLength]) {
                score[0]++;
            }
            if (answers[i] == b[i%bLength]) {
                score[1]++;
            }
            if (answers[i] == c[i%cLength]) {
                score[2]++;
            }
        }

        List<Integer> answerList = new ArrayList<>();
        int maxScore = Math.max(Math.max(score[0], score[1]), score[2]);
        if (maxScore == score[0]) {
            answerList.add(1);
        }
        if (maxScore == score[1]) {
            answerList.add(2);
        }
        if (maxScore == score[2]) {
            answerList.add(3);
        }

        int answerSize = answerList.size();
        int[] answer = new int[answerSize];
        for (int i = 0; i < answerSize; i++) {
            answer[i] = answerList.get(i);
        }
        return answer;
    }
}
